DOCS-9998-sftp-operations-editorial-kt (#1255)
mulesoft/docs-connectors: sftp/1.3/modules/ROOT/edits/sftp-copy-move.adoc

= Copy and Move Files with the SFTP Connector - Mule 4
:page-aliases: connectors::sftp/sftp-copy-move.adoc

Anypoint Connector for SFTP (SFTP Connector) Copy and Move operations enable you to copy and move files or directories on demand.

> While SFTP has a native concept of copying and moving, FTP does not. FTP Connector simulates these operations by downloading file content and writing it to a new location. Although all of this is done with concurrent streams that start writing and reading at the same time, copying and moving files does have a significant network overhead and should be used wisely. The SFTP Connector Move operation does not impact network overhead.

Reviewer: I'm not sure about "having a concept" of actions (as opposed to an idea). Maybe...
```suggestion
While using SFTP includes copying and moving operations, FTP does not. FTP Connector simulates these operations by downloading file content and writing it to a new location, using concurrent streams that start writing and reading at the same time. Because these operations create significant network overhead, use them wisely. Note that the SFTP Connector Move operation does not impact network overhead.
```

== Configure the Copy Operation in Studio

To add and configure the Copy operation in Studio, follow these steps:

. In the *Mule Palette* view, search for `sftp` and select the *Copy* operation.
. Drag the *Copy* operation onto the Studio canvas.
. In the *General* tab of the operation configuration screen, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. Specify the connection information and click *OK*.

> . Set the *Source path* field to the path of the file to copy, for example `source.txt`.

Reviewer: comma
```suggestion
. Set the *Source path* field to the path of the file to copy, for example, `source.txt`.
```
You can always check this by looking up ", for example, " in the style ref.

> . In the *Target path* field, enter the directory path into which to copy the file, for example `backup`.

Reviewer: "into which" in this sentence is a bit awkward. How about...:
```suggestion
. In the *Target path* field, enter the path of the directory to receive the copied file, for example, `backup`.
```

> . Set the *Rename to* field to rename the file as part of the operation, for example `renamed.txt`. +

Reviewer: comma
```suggestion
. Set the *Rename to* field to rename the file as part of the operation, for example, `renamed.txt`. +
```

Reviewer: linebreak

The following screenshot shows the *Copy* operation configuration:

.Copy operation configuration
image::sftp-copy-operation.png[Copy operation configuration in Studio]

In the XML editor, the `<sftp:copy>` configuration looks like this:
[source,xml,linenums]
----
<sftp:copy sourcePath="source.txt"
  targetPath="backup" overwrite="true|false"
  createParentDirectories="true|false"
  renameTo="renamed.txt"/>
----

== Configure the Move Operation in Studio

To add and configure the Move operation in Studio, follow these steps:

. In the *Mule Palette* view, search for `sftp` and select the *Move* operation.
. Drag the *Move* operation onto the Studio canvas.
. In the *General* tab of the operation configuration screen, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. Specify the connection information and click *OK*.
. Set the *Source path* field to the path of the file to move, for example `source.txt`.
. In the *Target path* field, enter the directory path into which to move the file, for example `backup`.

> . Set the *Rename to* field to rename the file as part of the operation, for example `renamed.txt`. +

Reviewer: ditto previous edits: commas, "into which"
